docs/aiken-integration.md

```diff
@@ -0,0 +1,516 @@
+# Aiken Integration Report
+This document describes LambdaBuffers integration with Aiken.
+It appears at Aiken's best,
+while it is possible to provide limited support for a subset of LambdaBuffers' features,
+integration with LambdaBuffers would provide a poor user experience that would be in conflict with either the major language features of Aiken or the key functionalities provided by LambdaBuffers.
+As such, this document describes the challenges LambdaBuffers integration with Aiken,
+and proposes alternative Milestone 4 outputs to better foster adoption of LambdaBuffers within the Cardano ecosystem.
+
+## Aiken limitations
+
+This section describes limitations with Aiken.
+
+### Aiken has no type class support
+
+A key feature of LambdaBuffers is to provide both types and type class instances.
+Aiken has no support for type classes, so one must generate the type class system themselves.
+In other words, one must provide:
+
+* A calling convention for functions with type classes.
+
+* Functions to create instance dictionaries for instances of the type class.
+
+The former is straightforward to implement.
+One can explicit pass an instance dictionary for each function that requires a type class as per the usual compilation of type class code to type class free code.
+
+The latter requirement poses troubles for Aiken.
+Aiken does not allow one to create instance dictionaries (product types of the instance's methods) since composite types in Aiken cannot contain a function type.
+
+For example, the following Aiken type
+
+```ruost
+type EqDict<a> {
+    eq: fn(a,a) -> Bool, 
+    neq: fn(a,a) -> Bool, 
+}
+```
+
+would produce an error like
```

Comment:
Add somewhere which version of Aiken you've used please. it gives nice error message only on master, stable release crashes (as of v26)